frontend/src/components/management/WorkflowAssignment.tsx

```diff
@@ -0,0 +1,448 @@
+            setCommonProjects([])
+            setIsFetching(undefined)
+            setSelectedProject(undefined)
+            setSelectedStudyWorkflow(undefined)
+            setIsFetching(undefined)
+            return
+        }
+
+        setIsFetching("Loading pools...")
+        const pooledSamples = (await dispatch(api.pooledSamples.list({ sample__id__in: sampleIDs.join(',') }))).data.results
+        const projectsBySample = [...samples, ...pooledSamples].reduce<Record<Sample['id'], Set<Project['id']>>>((acc, sample) => {
+            if ('pool_id' in sample) {
+                // handle pooled samples
+                if (!acc[sample.pool_id]) {
+                    acc[sample.pool_id] = new Set()
+                }
+                acc[sample.pool_id].add(sample.project_id)
+            } else {
+                // handle non-pooled samples
+                if (!acc[sample.id]) {
+                    acc[sample.id] = new Set()
+                }
+                if (sample.project) {
+                    acc[sample.id].add(sample.project)
+                }
+            }
+            return acc
+        }, {})
+
+        setIsFetching("Loading common steps...")
+        type StudyStepOrderKey = `${FMSSampleNextStepByStudy['study']}-${FMSSampleNextStepByStudy['step_order']}`
+        const sampleNextStepByStudyList = (await dispatch(api.sampleNextStepByStudy.getStudySamples({ sample_next_step__sample__id__in: sampleIDs.join(','), limit: 10000 }))).data.results
+        const studyStepCount: Record<string, number> = {}
+        const sampleIDsByStudyStep: Record<StudyStepOrderKey, Set<Sample['id']>> = {}
+        for (const { study, step_order, sample } of sampleNextStepByStudyList) {
+            const key: StudyStepOrderKey = `${study}-${step_order}`
+
+            studyStepCount[key] = (studyStepCount[key] ?? 0) + 1
+
+            if (!sampleIDsByStudyStep[key]) {
+                sampleIDsByStudyStep[key] = new Set()
+            }
+            sampleIDsByStudyStep[key].add(sample)
+        }
+        type StudyStepPair = [FMSSampleNextStepByStudy['study'], FMSSampleNextStepByStudy['step_order']]
+        const commonStudySteps = Object.entries(studyStepCount).reduce<StudyStepPair[]>((acc, [key, count]) => {
+            if (count > 0) {
+                const [study, stepOrder] = key.split('-')
+                acc.push([parseInt(study), parseInt(stepOrder)])
+            }
+            return acc
+        }, [])
+
+        setIsFetching("Loading projects...")
+        const countByProjectID = Object.values(projectsBySample).reduce<Record<Project['id'], number>>((acc, projects) => {
+            for (const project of projects) {
+                acc[project] = (acc[project] ?? 0) + 1
+            }
+            return acc
+        }, {})
+        const projectIDs = Object.keys(countByProjectID).map((id) => parseInt(id))
+        const commonProjects = (await fetchProjects(projectIDs)).map((project) => ({
```

Comment:
we should be using fetchX more often across the app ... This may help reduce calls to the backend...

Reply:
The sad thing about fetch is it doesn't handle filtering...